components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -2045,6 +2045,20 @@ private String handleOAuthAuthorizationRequest(OAuthMessage oAuthMessage)
         if (redirectURI != null) {
             return redirectURI;
         }
+        // Check whether PAR should be mandated in  the request.
+        OAuthAppDO oAuthAppDO;
+        try {
+            oAuthAppDO = OAuth2Util.getAppInformationByClientId(params.getClientId(), params.getTenantDomain());
+        } catch (IdentityOAuth2Exception | InvalidOAuthClientException e) {
+            throw new InvalidRequestException(e.getMessage(), e.getErrorCode());
+        }
+        if (oAuthAppDO.isRequirePushedAuthorizationRequests()) {
+            if (!Boolean.parseBoolean(oauthRequest.getParam(OAuthConstants.IS_PUSH_AUTHORIZATION_REQUEST))) {
```

Comment:
Lets get this directly from the request
```suggestion
            if (Boolean.TRUE.equals(oAuthMessage.getRequest().getAttribute(OAuthConstants.IS_PUSH_AUTHORIZATION_REQUEST))) {
```

Reply:
fixed